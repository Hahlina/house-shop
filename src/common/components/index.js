export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
