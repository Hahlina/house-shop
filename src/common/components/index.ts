export * from './PropertyCard/PropertyCard';
export * from './PropertyCard/PropertyCardLoader';
export * from './SwiperSlide/SwiperSlide';
export * from './SwiperContainer/SwiperContainer';
export * from './NavBreadcrumbs/NavBreadcrumbs';
export * from './Container/Container';
export * from './Image/Image';
export * from './ErrorPage/ErrorPage';
export * from './SomethingWentWrong/SomethingWentWrong';
export * from './NothingFound/NothingFound';
