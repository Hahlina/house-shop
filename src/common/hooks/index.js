export * from './useProperties';
export * from './useProperty';
