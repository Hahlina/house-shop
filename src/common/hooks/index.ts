export * from './state';
export * from './useImageOnLoad';
