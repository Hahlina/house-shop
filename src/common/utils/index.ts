export * from './convertPrice';
export * from './capitalizeFirstLetter';
