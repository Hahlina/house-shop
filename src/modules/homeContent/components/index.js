export * from "./PreviewBanner/PreviewBanner";
export * from "./PropertiesList/PropertiesList";
