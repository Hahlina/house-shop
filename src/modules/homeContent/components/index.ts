export * from './PreviewBanner/PreviewBanner';
export * from './PropertiesList/PropertiesList';
export * from './PropertiesPagination/PropertiesPagination';
