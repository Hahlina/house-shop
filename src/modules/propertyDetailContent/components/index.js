export * from "./PropertyGallery/PropertyGallery";
export * from "./FullInformationCard/FullInformationCard";
