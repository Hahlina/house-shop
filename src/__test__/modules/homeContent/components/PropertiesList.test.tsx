import { PropertiesList } from 'modules/homeContent/components';

import { render, screen, mockProperties } from 'common/utils/test';
import * as reduxHooks from 'common/hooks';

jest.mock('common/hooks');

describe('PropertiesList component', () => {
    const mockedUseAppSelector = jest.spyOn(reduxHooks, 'useAppSelector');
    const mockedUseAppDispatch = jest.spyOn(reduxHooks, 'useAppDispatch');

    beforeEach(() => {
        mockedUseAppDispatch.mockReturnValue(jest.fn());
    });

    it('should render a list of properties when data is available', () => {
        mockedUseAppSelector.mockReturnValue({ data: mockProperties, loading: false });
        render(<PropertiesList />);
        expect(screen.getByText(mockProperties[0].title)).toBeInTheDocument();
    });

    it('should render a loading skeleton when data is not available and loading is true', () => {
        mockedUseAppSelector.mockReturnValue({ data: [], loading: true });
        render(<PropertiesList />);
        expect(screen.getByTestId('properties-list-loader')).toBeInTheDocument();
    });

    it('should match snapshots', () => {
        mockedUseAppSelector.mockReturnValue({ data: mockProperties, loading: false });
        const { container } = render(<PropertiesList />);
        expect(container).toMatchSnapshot();
    });
});
