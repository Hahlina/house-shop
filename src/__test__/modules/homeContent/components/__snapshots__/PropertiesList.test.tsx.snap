// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertiesList component should match snapshots 1`] = `
.c1 {
  width: 350px;
  height: 500px;
  border: 1px solid #979797;
  transition: box-shadow 0.3s ease;
}

.c1:hover {
  box-shadow: 0 34px 36px 0 #00000021;
}

.c2 {
  height: 280px;
}

.c3 {
  display: flex;
  flex-direction: column;
  height: 220px;
}

.c3>* {
  height: 33%;
}

.c3 p {
  font-size: 20px;
}

.c3 p.title {
  font-weight: 700;
}

.c3 p.price {
  color: #F4511E;
  font-size: 25px;
}

.c0 {
  margin: 100px 0;
}

.c0 p.list-title {
  position: relative;
  font-size: 36px;
  font-weight: 700;
  line-height: 64px;
  text-transform: capitalize;
  margin-bottom: 50px;
}

.c0 p.list-title::before {
  content: '';
  position: absolute;
  display: inline-block;
  height: 4px;
  width: 100px;
  background: #F4511E;
  bottom: 0;
}

.c0 div.card-list {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  gap: 30px 30px;
}

<div>
  <div
    class="c0"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 list-title css-ahj2mt-MuiTypography-root"
    >
      List of properties
    </p>
    <div
      class="card-list MuiBox-root css-0"
    >
      <a
        to="property/1"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c1 css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root c2 css-16cvb1l-MuiCardMedia-root"
              role="img"
              style="background-image: url(image.jpg);"
            />
            <div
              class="MuiCardContent-root c3 css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
              >
                Beautiful House
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 address css-ahj2mt-MuiTypography-root"
              >
                3 Main Staks
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 price css-ahj2mt-MuiTypography-root"
              >
                $100,000.00
              </p>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </a>
      <a
        to="property/2"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c1 css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root c2 css-16cvb1l-MuiCardMedia-root"
              role="img"
              style="background-image: url(image_2.jpg);"
            />
            <div
              class="MuiCardContent-root c3 css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
              >
                Less Beautiful House
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 address css-ahj2mt-MuiTypography-root"
              >
                123 Main St
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 price css-ahj2mt-MuiTypography-root"
              >
                $6,600,000.00
              </p>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </a>
      <a
        to="property/3"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c1 css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <button
            class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiCardMedia-root c2 css-16cvb1l-MuiCardMedia-root"
              role="img"
              style="background-image: url(image_2.jpg);"
            />
            <div
              class="MuiCardContent-root c3 css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 title css-ahj2mt-MuiTypography-root"
              >
                More Beautiful House
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 address css-ahj2mt-MuiTypography-root"
              >
                USA
              </p>
              <p
                class="MuiTypography-root MuiTypography-body1 price css-ahj2mt-MuiTypography-root"
              >
                $300,000.00
              </p>
            </div>
            <span
              class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </a>
    </div>
  </div>
</div>
`;
